### ISSUE TITLE
v.0.1.5 - unable to complete Rx subscription after update.

### ISSUE THREAD

**Issue by bovquier** [koral--/gradle-pitest-plugin#30]
While doing pitests from test:
```
def "should do the job"() {
        given:
        presenter.setModel(model)

        and:
        presenter.doTheJob(view)
        testScheduler.triggerActions()

        assert presenter.answersSet.asList() == [1, 2, 3]

        when:
        presenter.optionSelected(4, true)

        then:
        presenter.answersSet.asList() == [1, 2, 3, 4]
    }
```

which runs on methods from presenter:

```
Set<Integer> answersSet = new TreeSet<>();
    InputAndAnswerEntity answerEntity;

    @Override
    public void setModel(InputAndAnswerEntity input) {
        this.model = input;
        this.answerEntity = input;
    }

    public void doTheJob(MultiSelectionContract.View view) {
        Subscription subscribe = Observable
                .from(getOptionsList(answerEntity))
                .subscribeOn(observeOn)
                .observeOn(observeOn)
                .subscribe(
                        view::addOption,
                        trackingController::trackException,
                        () -> {
                            answersSet = InputAndAnswerEntity.getAnswerAsSet(answerEntity);
                            for (Integer answer : answersSet) {
                                view.selectOption(answer);
                            }
                        }
                );
        compositeSubscription.add(subscribe);
    }
    private void storeAnswer() {
        JSONArray jsonArray = new JSONArray();
        for (Integer answer : answersSet) {
            jsonArray.put(answer);
        }
        answerEntity.setAnswer(jsonArray.toString());
        repositoryController.storeInputAndAnswerEntity(answerEntity);
    }

    void readAnswer() {
        answersSet = InputAndAnswerEntity.getAnswerAsSet(answerEntity);
    }

    @Override
    public void optionSelected(int checkedId, boolean isChecked) {
        if (isChecked) {
            answersSet.add(checkedId);
        } else {
            answersSet.remove(checkedId);
        }
        storeAnswer();
    }
```

it results in:

```
1:03:52 PM PIT >> INFO : MINION : 1:03:52 PM PIT >> SEVERE : Description [testClass=SelectionPresenterSpec, name=SelectionPresenterSpec]
Condition not satisfied:

presenter.answersSet.asList() == [1, 2, 3]
|         |          |        |
|         []         []       false
SelectionPresenter@6e6e39e2 
	at SelectionPresenterSpec.should do the job(SelectionPresenterSpec.groovy:168)

```

this happens after updating to `0.1.5`, on `0.1.4` works good.
This looks like the update blocks Rx `subscribe` from going into `onCompleted` methods, because other test, which check for method invocations, also fails.

**Comment by koral--** [comment]
Sorry for the delay. 
Could you share a project containing minimal reproducible sample?
Samples you provided are incomplete e.g. it is unknown what is `observeOn` and where `compositeSubscription` is unsubscribed.

**Comment by bovquier** [comment]
https://github.com/bovquier/PiTestError

here you can find implementation that shows the error.

**Comment by koral--** [comment]
Thanks, I'll investigate that.

**Comment by koral--** [comment]
Fixed in version `0.1.7-SNAPSHOT`. However, you also need to add those lines to `buildscript` block in root `build.gradle` file:
`configurations.maybeCreate("pitestTestCompile")`
and inside `dependencies`:
`pitestTestCompile "org.json:json:20140107"` (version is taken from your sample project).

TL;DR
It worked in `0.1.4` but stopped working in `0.1.5` due to fix for #24.
You are using classes from `org.json:json` which are also provided by Android framework (mockable JAR in tests). 
Mockable JAR has to be located before at least test classes in classpath to work correctly. Before `0.1.7`  `pitestTestCompile` configuration was used to provide that JAR. Adding just `pitestTestCompile "org.json:json:20140107"` won't fix this issue before `0.1.7` because order of configuration elements is undefined but json has be to be located before mockable JAR.

Now mockable JAR is not added to `pitestTestCompile` but included separately after it.
